### PR TITLE
Add digital filters to the ADC data acquisition flow

### PR DESCRIPTION
**Pull request by adisuciu**
Filters are only enabled for lowgain mode, as high gain should be calibrated using the trimmers on the M2k. There is an option to show/hide the controls in the oscilloscope channels. The values for the filters are saved in the Scopy.ini file.

### CODE REVIEW

**Comment by adisuciu** [comment]
Filters are now available for both low and high gain mode with different configurations

File: src/frequency_compensation_filter_impl.cc [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+			out[i]=in[i]+(short)((out_f[i])*config[high_gain].gain);
+		}
+
+	}
+	else
+	{
+		memcpy(out,in,noutput_items*sizeof(short));
+	}
+	delete []out_f;
+        return noutput_items;
```
Comment (AlexandraTrifan) [inline src/frequency_compensation_filter_impl.cc]: Indentation looks a bit different when comparing to other source files. Can we astyle the newly added files?

File: src/frequency_compensation_filter.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+namespace adiscope {
+
+    class frequency_compensation_filter : virtual public gr::sync_block
+    {
+    public:
+      // gr::blocks::streams_to_vector::sptr
+      typedef boost::shared_ptr<frequency_compensation_filter> sptr;
+
+      /*!
+       * Make a stream-to-vector block.
```
Comment (AlexandraTrifan) [inline src/frequency_compensation_filter.h]: Is this comment related to the GR block?

File: src/oscilloscope_api.cpp [hunk 3]
```diff
@@ -639,6 +639,17 @@ QVariantList Oscilloscope_API::getChannels()
 	return list;
 }
 
+QVariantList Channel_API::getDigFilters() const
+{
+	QVariantList list;
+
+	for (Channel_Digital_Filter_API *each : digFilters)
```
Comment (AlexandraTrifan) [inline src/oscilloscope_api.cpp]: missing "{" and  "}" ?

File: src/oscilloscope_api.hpp [hunk 4]
```diff
@@ -308,10 +343,21 @@ class Oscilloscope_API : public ApiObject
...
+
 public:
 	explicit Channel_API(Oscilloscope *osc) :
-		ApiObject(), osc(osc) {}
-	~Channel_API() {}
+		ApiObject(), osc(osc) {
+			for(auto i=0;i<2;i++)
+				digFilters.append(new Channel_Digital_Filter_API(osc,this,i));
+		}
+
```
Comment (AlexandraTrifan) [inline src/oscilloscope_api.hpp]: missing "{" and "}" ?

File: src/frequency_compensation_filter_impl.cc (hunk 1 above)
Comment (adisuciu) [reply src/frequency_compensation_filter_impl.cc]: fixed

File: src/frequency_compensation_filter.h (hunk 2 above)
Comment (adisuciu) [reply src/frequency_compensation_filter.h]: fixed

File: src/oscilloscope_api.cpp (hunk 3 above)
Comment (adisuciu) [reply src/oscilloscope_api.cpp]: fixed

File: src/oscilloscope_api.hpp (hunk 4 above)
Comment (adisuciu) [reply src/oscilloscope_api.hpp]: fixed
